docs/developer/architecture/user_customisations.rst

```diff
@@ -0,0 +1,59 @@
+User Customisations and Persistence
+===================================
+
+Context
+-------
+
+MetalK8s comes with a set of tools and services that may need to be configured
+on site. At the same time, we don't want the administrator of the cluster to
+master each and every service of the cluster. We also don't want to allow all
+kind of customisations since it would make the system even more complex to test
+and maintain over time.
+
+In addition to those services, MetalK8s deployment may have to be adapted
+depending on the architecture of the platform or depending on the different
+use cases and applications running on the top of it.
+
+It can be:
+
+- The BootstrapConfig,
+- The various roles and taints we set on the node objects of the cluster
+- The labels and annotations we would set on all K8s objects
+- The configurations associated to solutions, such as the list of available
+  solutions, the environments and namespaces created for a solution
+
+Be it services or MetalK8s customisations, we need to ensure it is persisted
+and resistant to various type of events such as node reboot, upgrade, downgrade
+, backup, restore.
+
+.. _usercusto-requirements:
+
+Requirements
+------------
+
+As a MetalK8s administrator, I have access to a finite list of settings I can
+customise on-site in order to match with my environment specificities:
+
+- list of static users and credentials configured in Dex
+- Integration with an external IDP configuration in Dex
+- Alert notification configuration in Alert Manager
+- Number of replicas for the Prometheus, Alert Manager, Grafana or Dex
+- Other items will appear as we add new customisable features in MetalK8s
+
```

Comment:
It's a question from UI part.
- Currently, we store the language preference in local storage.
- We are going to support light/dark theme switch very sson.
Wondering do we need to take these settings(language/theme) into consideration?

Reply:
Good question. I think we're not talking about the same topics however, so no, shouldn't be in here.
Calling this feature "User customizations" is actually misleading: there's nothing related to a user here, we're talking about "Cluster services configuration".
In the UI case however, this is indeed "User preferences". It could warrant a dedicated architecture document (cc @thomasdanan), but do you think `LocalStorage` isn't a good design for it @ChengYanJin?

Reply:
Yeah. it's good to store the settings in Localstorage!
Maybe we will want to store the "user preference" during the upgrade as well?

Reply:
I'd also like users to be able to add their own alerts to Prometheus. Differing customer workloads often surface new/unknown bottlenecks, issues, etc that we may need to monitor. Users or the CSE should be able to easily and persistently edit the alerts.

Reply:
Good point @bkettler, will add this to the list of topics.